lib/avro/resolution_canonical_form.rb

```diff
@@ -12,20 +14,47 @@ def to_resolution_form(schema)
 
     private
 
-    # TODO: include aliases once the avro Ruby gem supports them
-    # Note: permitted values for defaults are not enforced here.
-    # That should be done at the schema level, and is not done currently
-    # in the Avro Ruby gem
+    def normalize_schema(schema)
+      if schema.type_sym == :bytes && schema.logical_type == DECIMAL_LOGICAL_TYPE
+        add_logical_type(schema, type: schema.type)
+      else
+        super
+      end
+    end
+
     def normalize_field(field)
-      default_value = if field.default?
-                        { default: field.default }
-                      else
-                        {}
-                      end
-      super.merge(default_value)
+      extensions = {}
+      extensions[:default] = field.default if field.default?
+      if field.respond_to?(:aliases) && field.aliases && !field.aliases.empty?
+        extensions[:aliases] = field.aliases.sort
+      end
+
+      super.merge(extensions)
     end
 
-    # TODO: Override normalize_named_type once the Avro Ruby gem supports aliases
-    # def normalized_named_type(schema, attributes = {})
+    def add_logical_type(schema, serialized)
+      if schema.respond_to?(:logical_type) && schema.logical_type == DECIMAL_LOGICAL_TYPE
+        extensions = { logicalType: DECIMAL_LOGICAL_TYPE }
+        extensions[:precision] = schema.precision if schema.respond_to?(:precision) && schema.precision
```

Comment:
We need the `respond_to?` checks on this line and the next line because decimals can be `Avro::Schema::BytesSchema` or `Avro::Schema::FixedSchema` right? Do we need to handle the `Avro::Schema::FixedSchema#size` attribute?

Reply:
This fixed `size` attribute is already handled as part of the standard schema normalization: https://github.com/apache/avro/blob/master/lang/ruby/lib/avro/schema_normalization.rb#L54

And yes, we need the `respond_to?` checks because the fixed type supports the decimal logical type in the Avro spec, but this is not yet implemented in the Ruby library. The other option would to check that the `logical_type` is `decimal` and the `type` is `bytes` for now. I implemented it this way so that no update would be required to support fixed decimals once the Ruby implementation catches up.